pulsar-broker/src/test/java/org/apache/pulsar/client/impl/BrokerClientIntegrationTest.java

```diff
@@ -394,6 +394,10 @@ public void testResetCursor(SubscriptionType subType) throws Exception {
 
         final NavigableMap<Long, TimestampEntryCount> publishTimeIdMap = new ConcurrentSkipListMap<>();
 
+        // set delay time to start dispatching messages to active consumer in order to avoid message duplication
```

Comment:
Can you also add a test specific for this feature, maybe adding multiple consumers in short sequence (before the delay expires).

Reply:
I've added the test to PersistentFailoverE2ETest.